### DISCUSSION TITLE
What is the "AppUserModelID "?

### DISCUSSION THREAD

**Discussion by qalisto** [espanso/espanso#1818] [Q&A]
Per the below log, it is throwing an error in my Win11 setup.
Not clear on the _"unable to find espanso AppUserModelID in the list of registered ones, falling back to Powershell"_ implications for Usage of Espanso.  App seems to be operating.

What is the AppUserModelID and ... how do we determine what is "Registered"?

Direction appreciated.

09:17:38 [daemon(14752)] [INFO] reading configs from: "e:\\0-Z_DT_32g\\E\\ESO-PORT\\.espanso"
09:17:38 [daemon(14752)] [INFO] reading packages from: "e:\\0-Z_DT_32g\\E\\ESO-PORT\\.espanso\\match\\packages"
09:17:38 [daemon(14752)] [INFO] using runtime dir: "e:\\0-Z_DT_32g\\E\\ESO-PORT\\.espanso-runtime"
09:17:38 [daemon(14752)] [INFO] system info: Windows v11 (22631) - kernel: 22631
09:17:38 [daemon(14752)] [INFO] watching for changes in path: "e:\\0-Z_DT_32g\\E\\ESO-PORT\\.espanso"
09:17:38 [daemon(14752)] [INFO] espanso version: 2.2.1
09:17:38 [daemon(14752)] [INFO] spawning the worker process...
09:17:38 [daemon(14752)] [INFO] binded to named pipe: \\.\pipe\espansodaemonv2
09:17:38 [worker(25056)] [INFO] reading configs from: "e:\\0-Z_DT_32g\\E\\ESO-PORT\\.espanso"
09:17:38 [worker(25056)] [INFO] reading packages from: "e:\\0-Z_DT_32g\\E\\ESO-PORT\\.espanso\\match\\packages"
09:17:38 [worker(25056)] [INFO] using runtime dir: "e:\\0-Z_DT_32g\\E\\ESO-PORT\\.espanso-runtime"
09:17:38 [worker(25056)] [INFO] system info: Windows v11 (22631) - kernel: 22631
09:17:38 [worker(25056)] [INFO] binded to named pipe: \\.\pipe\espansoworkerv2
09:17:38 [worker(25056)] [INFO] using Win32AppInfoProvider
09:17:38 [worker(25056)] [INFO] monitoring the status of the daemon process
09:17:38 [worker(25056)] [INFO] using Win32Source
09:17:38 [worker(25056)] [INFO] using Win32Injector
09:17:38 [worker(25056)] [INFO] using Win32Clipboard
09:17:42 [worker(25056)] [WARN] unable to find espanso AppUserModelID in the list of registered ones, falling back to Powershell
09:20:10 [worker(25056)] [WARN] rendering was aborted by extension: form, on var: form1

**Comment by smeech** [comment]
I'm not an expert, particularly on Windows, but is it related to the Taskbar icon?
These files may be relevant:
https://github.com/espanso/espanso/blob/3f6444c8b5c0004287b9deb6b6a1aa414b1a36e4/espanso-ui/src/win32/notification.rs#L52
and
https://github.com/espanso/espanso/blob/3f6444c8b5c0004287b9deb6b6a1aa414b1a36e4/scripts/resources/windows/setupscript.iss#L9
Hopefully someone who knows what they're talking about will jump in!

**Reply by qalisto** [reply]
I hope someone jumps in too ... especially if this situation is exposing Security risks.  Meantime, it is working.  And we are grateful to you and Federico and the other Collabs who are sustaining this important Utility.
